vike/client/client-routing-runtime/prefetch.ts

```diff
@@ -95,17 +170,26 @@ function addLinkPrefetchHandlers(pageContext: { exports: Record<string, unknown>
 
     if (isAlreadyPrefetched(url)) return
 
-    const { prefetchStaticAssets } = getPrefetchSettings(pageContext, linkTag)
-    if (!prefetchStaticAssets) return
+    const { prefetchStaticAssets, prefetchPageContext } = getPrefetchSettings(pageContextBeforeRenderClient, linkTag)
+    if (!prefetchStaticAssets && !prefetchPageContext) return
+
+    const pageContext = await createPageContext(url)
+    let pageContextFromRoute: PageContextFromRoute
+    try {
+      pageContextFromRoute = await route(pageContext)
```

Comment:
I don't think we should route here: I'm inclined to call the users's Route Functions not eagerly and only when strictly needed.

We can share the routing result by merging the event listeners.

Reply:
> We can share the routing result by merging the event listeners.

I'm not sure I understand, does this mean merging `route()` into the logic of `addLinkPrefetchHandlers()`?

Reply:
I think we can merge `prefetchContextIfPossible()` and `prefetchAssetsIfPossible()` into one function and call `route()` in there. That way we call `route()` only if necessary / when we need to, and only once.

Reply:
I might not fully understand your point yet. `The route()` is not called every time a hover event occurs, but only when the page is loaded. 
Are you suggesting that you want to reduce the number of times route() is called? 
If we merge `prefetchContextIfPossible()` and `prefetchAssetsIfPossible()` into a single function and include the `route()` function, it would result in calling it every time a hover event occurs, which seems to be the scenario you are concerned about.

Reply:
> `The route()` is not called every time a hover event occurs, but only when the page is loaded.

That's probablematic in dev, as it's essentially a cache that breaks HMR.

> it would result in calling it every time a hover event occurs

I think it's fine, that's how it was working before this PR and no one complained.

> I might not fully understand your point yet.

It's just about calling `route()` one time instead of two times per hover.

Reply:
> That's probablematic in dev, as it's essentially a cache that breaks HMR.

I understand now that it's not correct for `route()` to be called only when the page is loaded. Thank you for the clarification.

> I think we can merge prefetchContextIfPossible() and prefetchAssetsIfPossible() into one function and call route() in there.

I've implemented it now.